ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -77,16 +80,20 @@
   private final StoreKeyFactory factory;
   private final File bloomFile;
   private final StoreMetrics metrics;
-  // an approximation of the last modified time.
-  private final AtomicLong lastModifiedTimeSec;
   private final AtomicInteger numberOfItems;
 
+  // an approximation of the last modified time.
+  private AtomicLong lastModifiedTimeSec;
```

Comment:
Incase of initializing an IndexSegment from file, we populate the lastModifiedTime only when we read the field from the file. Before this patch, we just assign to the file attribute "indexFile.lastModified()". And hence removed the final